frame/staking/src/pallet/mod.rs

```diff
@@ -559,6 +646,179 @@ pub mod pallet {
 	#[pallet::getter(fn current_planned_session)]
 	pub type CurrentPlannedSession<T> = StorageValue<_, SessionIndex, ValueQuery>;
 
+	/// Wrapper struct for Era related information. It is not a pure encapsulation as these storage
+	/// items can be accessed directly but nevertheless, its recommended to use `EraInfo` where we
+	/// can and add more functions to it as needed.
+	pub(crate) struct EraInfo<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> EraInfo<T> {
+		/// Temporary function which looks at both (1) passed param `T::StakingLedger` for legacy
```

Comment:
Not sure but maybe this impl should be under `impls.rs` instead? Also not a big fan of this dir org as we discussed, but for now I'd keep consistency.

Reply:
Since this is mostly working with storage items, perhaps it can stay here. `impls.rs` is mainly implying _trait implementations_.